node-notes-cli/app.js

```diff
@@ -0,0 +1,74 @@
+import * as fs from 'node:fs/promises';
+
+async function readNotes() {
+  try {
+    const fileContent = await fs.readFile('data.json', 'utf-8');
+    const data = JSON.parse(fileContent);
+    for (const key in data.notes) {
+      console.log(`${key}: ${data.notes[key]}`);
+    }
+  } catch (err) {
+    console.error(err);
+    process.exit(1);
+  }
+}
+
+async function createNote() {
+  try {
+    const newNote = process.argv[3];
+    const fileContent = await fs.readFile('data.json', 'utf8');
+    const data = JSON.parse(fileContent);
+    const newNoteId = data.nextId;
+    data.notes[newNoteId] = newNote;
+    data.nextId++;
+    await fs.writeFile('data.json', JSON.stringify(data, null, 2));
```

Comment:
You repeat this `writeFile` call several times. You might want to encapsulate it into a function, as well.

One advantage of encapsulating into functions is that you don't have to type so much. Another is that if you find a bug or change your mind about the implementation, you only have to change one place in the code. For example, think about all the changes you'd need to make to this file if you decided to use a database for your notes instead of the file system.